index.bs

```diff
@@ -76,773 +93,795 @@ All <{iframe}> use cases are still valid and should continue to use <{iframe}>,
-};
+-  Only embedder applications and their same-origin [[Isolated-Web-Apps|IWA]]
+    child navigables that have been granted the "controlledframe"
+    policy-controlled feature will be allowed access to the Controlled Frame
+    element.
+-  Same-origin child navigables without the "controlledframe"
+    policy-controlled feature will not be allowed to use a Controlled Frame
+    element (to be confirmed, see note below)
+-  Cross-origin iframes to [[Isolated-Web-Apps|IWAs]] can use the "allow"
+    attribute to disable Controlled Frame
+-  Cross-origin iframes to non-[[Isolated-Web-Apps|IWAs]] will not have
+    access to the "controlledframe" policy-controlled feature, and so will not
+    be allowed to use a Controlled Frame element
+-  See the "Embedder Policies Using the "controlledframe" Feature" section
+    above for more details
 
-dictionary HttpHeader {
-  required DOMString name;
-  DOMString value;
-  sequence<byte> binaryValue;
-};
+<b>An IWA may attempt to embed content from non-https schemes, such as 'http:'
+or 'isolated-app:'</b>
 
-dictionary WebRequestAuthCredentials {
-  required DOMString username;
-  required DOMString password;
-};
+Planned mitigation:
 
-dictionary BlockingResponse {
-  WebRequestAuthCredentials authCredentials;
-  boolean cancel;
-  USVString redirectUrl;
-  sequence<HttpHeader> requestHeaders;
-  sequence<HttpHeader> responseHeaders;
-};
+-  Controlled Frame will only work when the navigable's "src" URL has an
+    'https:' scheme
 
-enum DocumentLifecycle {
-  "prerender",
-  "active",
-  "cached",
-  "pending_deletion",
-};
+<b>Malicious Controlled Frame could access the embedder's running process (eg.
+Spectre attack)</b>
 
-enum FrameType {
-  "outermost_frame",
-  "fenced_frame",
-  "sub_frame",
-};
+Planned mitigation:
 
-dictionary WebRequestEventDetails {
-  DOMString documentId;
-  DocumentLifecycle documentLifecycle;
-  required long frameId;
-  FrameType frameType;
-  USVString initiator;
-  required DOMString method;
-  DOMString parentDocumentId;
-  required long parentFrameId;
-  required DOMString requestId;
-  required long timeStamp;
-  required ResourceType type;
-  required USVString url;
-};
+-  Controlled Frame will be executed in a separate process from the
+    embedder's process
 
-dictionary WebRequestEventResponseDetails : WebRequestEventDetails {
-  required long statusCode;
-  required DOMString statusLine;
-  sequence<HttpHeader> responseHeaders;
-};
+<b>Controlled Frame for a given "https origin" could interact or interfere with
+the user's own storage user agent data for that https origin</b>
 
-dictionary UploadData {
-  ArrayBuffer bytes;
-  DOMString file;
-};
-dictionary RequestBody {
-  DOMString error;
-  any formData;
-  sequence<UploadData> raw;
-};
+Planned mitigation:
 
-dictionary WebRequestBeforeRequestDetails : WebRequestEventDetails {
-  RequestBody requestBody;
-};
+-  Controlled Frame will always store data in a separate storage user agent
```

Comment:
Is "storage user agent" an existing concept? I'm not sure how to best spec this. Maybe by saying we're adding an additional key to StorageKey similar to how we do it in IsolatedContext?

Reply:
OK, I tried something here. Let me know what you think!